Adding RETC NSIS install script

madalee-com opened the pull request:
Creating pull request to add NSI script file for NSIS installer.

This script requires NSIS to be installed, as well as the RETC and NSIS binaries, and RzChromaSDK and CorsairSDK DLLs to be placed in appropriate paths, before compiling the script.

Commit:
Adding RETC NSIS install script

Commit:
Renamed installer directory and added stub directories for binary and DLL files

Commit:
Changes round 1

MartB commented:
Looks like we resolved all the important issues.
:+1: from me, will merge this into develop tomorrow after i wake up if you have no more things to change.

MartB commented:
Review status of this merge:
https://github.com/MartB/RETC/issues/6#issuecomment-307481181

(just for reference)

Commit:
Several updates

- Fixed uninstall not removing registry keys
- Added alert during install if already installed
- Added check to System RzChromaSDK check to only offer uninstall if DLL ProductName is empty(prevents removal of Razer official SDK files)

**IMPORTANT**
MoreInfo plugin is now required to compile RETC installer
http://nsis.sourceforge.net/MoreInfo_plug-in

Commit:
Fixes

- Changed RzChromaSDK files to be installed to a wrappers subdirectory.  These cause a loop if they are in the same directory as the EXE.
- Added a check, during install, for the presence of RzChomeSDK in the RETC folder.  Installer will delete these files from the folder if found during install.
- Fixed issue with nssm being pointed to the 64 bit dll on 32 bit systems

Commit:
More uninstaller info

Added/modified URLUpdateInfo, URLInfoAbout, HelpLink, and EstimatedSize for the windows Add/Remove screen.

Commit:
Cleanup and Version

- Did a little cleanup and re-organization of the installer code
- Passing "/DVERSION=1.2.3.4" to makensis will allow setting the version number seen on Add/Remove programs.  Obviously you should replace 1.2.3.4 with the version number.
- Added commented out section to default to a getting the version number from the RETC EXE file.  The file currently doesn't have a version number, so this will cause makensis to abort, can uncomment if version number is added to RETC EXE

Commit:
More add/remove info

- Added Publisher to add/remove
- Enabled code to detect version number from EXE

Commit:
EXE strings and log

- Fix issue with getting the version string from the EXE
- Added RETC as a ProductName string to identify the RETC version of the RzChromaSDK files
- Temporarily made the Install Log screen not jump to Finish, so we can get a better look at the log

Commit:
Debugging Install Log

- Added OEM option to nsExec lines to attempt to resolve log display issue on some systems

Commit:
Enable auto finish page

- Re-enabled jumping to finish page once install completes

MartB commented:
Merging this right now.
#merged
